ods_ci/run_interop.sh

```diff
@@ -0,0 +1,424 @@
+  for venvdir in "${venvdir}" "master"; do
+    virtenv="${HOME}/.local/ods-ci/${venvdir}/.venv"
+    echo "Checking whether '${virtenv}' exists."
+    if [[ -d "${virtenv}" ]]; then
+      break
+    fi
+  done
+  if [[ -d "${basepath}/../.venv" ]]; then
+    echo "We ran already before and settled in venv we linked into '${basepath}/../.venv'."
+  elif [[ -d "${virtenv}" ]]; then
+    echo "Using a pre-created virtual environment in '${virtenv}' for poetry to save time."
+    poetry config --local virtualenvs.in-project true
+    # Workaround for Poetry urllib3 connection error
+    poetry config --local installer.parallel false
+    # Link in the venv we're going to be using
+    ln --symbolic "${virtenv}" "${basepath}/../.venv"
+    # Workaround for .venv directory not being movable (hashbangs in bin/ hardcode python interpreter path)
+    ln --symbolic "${virtenv}" "${HOME}/.local/ods-ci/.venv"
+  else
+    echo "Pre-created virtual environment has not been found in '${virtenv}'. All dependencies will be installed from scratch."
+  fi
+  # ensure python 3.11
+  python=$(poetry env info --executable)
+  if [[ -n "${python}" ]] && ${python} -c 'import sys; sys.exit(0 if sys.version_info[0:2] == (3, 11) else 1)'; then
+    echo "Python '${python}' will be used"
+  else
+    echo "Python '${python}' is not of the correct version"
+    python311=$(which python3.11)
+    if [[ -n "${python311}" ]]; then
+      echo "Configuring poetry to use Python ${python311}"
+      poetry env use "${python311}"
+    else
+      echo "[ERROR] Python 3.11 was not found!"
+      echo "Install Python 3.11 on your machine. On Fedora, do 'sudo dnf install -y python3.11-devel'"
+      echo "then run 'poetry env use /path/to/python3.11' and then try running robot again"
+      exit 1
+    fi
+  fi
+
+  poetry install --no-interaction --sync || {
+      echo "[ERROR] poetry environment setup failed, aborting the execution!"
+      exit 1
+  }
+fi
+# shellcheck disable=SC1091
+source "$(poetry env info --path)/bin/activate"
+
+#Create a unique directory to store the output for current test run
+if [[ ! -d "${TEST_ARTIFACT_DIR}" ]]; then
+  mkdir -p "${TEST_ARTIFACT_DIR}"
+fi
+if ! ${SUBFOLDER}; then
+  case "$(uname -s)" in
+    Darwin)
+        # shellcheck disable=SC2046
+        TEST_ARTIFACT_DIR=$(mktemp -d  "${TEST_ARTIFACT_DIR}" -t "${TEST_ARTIFACT_DIR}"/ods-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+         ;;
+    Linux)
+        # shellcheck disable=SC2046
+        TEST_ARTIFACT_DIR=$(mktemp -d -p "${TEST_ARTIFACT_DIR}" -t ods-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+        ;;
+  esac
+fi
+
+
+robot ${TEST_EXCLUDE_TAG} ${TEST_INCLUDE_TAG} -d ${TEST_ARTIFACT_DIR} -x xunit_test_result.xml -r test_report.html ${TEST_VARIABLES} --variablefile ${TEST_VARIABLES_FILE} --exclude TBC ${EXTRA_ROBOT_ARGS} ${TEST_CASE_FILE}
+# shellcheck disable=SC2116
+exit_status=$(echo $?)
+echo "${exit_status}"
+
+# send test artifacts by email
+if ${EMAIL_REPORT}
```

Comment:
Sending test results should not be in scope of test image responsibility.
The test image should be responsible only for producing test results.
Further processing of test results is CI system's responsibility.

Reply:
Will remove the section for sending test results.